pyproject.toml

```diff
@@ -223,6 +224,7 @@ dask = "ibis.backends.dask"
 datafusion = "ibis.backends.datafusion"
 druid = "ibis.backends.druid"
 duckdb = "ibis.backends.duckdb"
+flink = "ibis.backends.flink"
```

Comment:
Since flink isn't a full-fledged backend yet, we should revert this entrypoint registry. Doing this should also let you revert the changes in `ibis/__init__.py` and `gen_matrix.py`.